File: EigenFaces, GMeans, PCA, BayesianNetworks, Validation/python/Q_4.py
from rpy2.robjects import r;

def bayesian (input):
    r('library(bnlearn, quietly=TRUE)');
    r('input = "'+input+'"');
    r('data = read.csv(input)');
    r('data = data[, !(colnames(data) %in% c("X"))]');
    r('res = hc(data)');
    r('png("Q4.png")')
    r('plot(res)');
    r('dev.off()');
    r('fit = bn.fit(res, data=data)');
    r('print(fit)');

File: EigenFaces, GMeans, PCA, BayesianNetworks, Validation/R/Q4.R
library(bnlearn, quietly=TRUE);

bayesian <- function (input) {
	data = read.csv(input);
	data = data[, !(colnames(data) %in% c("X"))];
	res = hc(data);

	return(res$nodes[1]);
	plot(res);
	
	fit = bn.fit(res, data=data);
	return(fit);
}